notewhacker.lisp

```diff
@@ -407,18 +417,26 @@ difficulty. Return the new target."
       ;; TODO: Better choice at random notes. Learning? (Make player try
       ;; and hit difficult notes, which player had problems with
       ;; earlier.)
-      (let ((base-key (if (eq (clef staff) 'f-clef)
-                          30
-                          55
-                          )))
+      (let* ((base-key (min-base-key staff))
+             (staff-width (- (max-base-key staff) base-key))
+             (max-chord-width 12) ; octave
+             (half-chord-width (/ max-chord-width 2)))
```

Comment:
Function `/` returns ratios in some cases (e.g. could be `11/2` if `max-chord-width` was 11). This is a problem later on because a ratio may be added to the list of notes to create (`chord-base-note` and hence `random-note` returns a ratio) and it goes ultimately to `aref` index in `position-and-accidental-of-mkn` in `notation.lisp`.

Instead of `/`, `floor` or `ceiling` could be used:
```suggestion
             (half-chord-width (floor max-chord-width 2)))
```

(I should have added `type-check`s and type annotations all over the place in the original code...)